### PR TITLE
Remove PlanVariableAllocator

### PR DESCRIPTION
**Pull request by jainxrohit**
Currently, both PlanVariableAllocator and VariableAllocator are responsible for similar tasks, leading to redundancy in the codebase. To address this, we are removing PlanVariableAllocator and consolidating all functionality within VariableAllocator.

Commit1 moves various methods to create variables from PlanVariableAllocator to  VariableAllocator.
Commit2 moves methods which relies on expression and fields from PlanVariableAllocator to PlannerUtils, which eventually uses VariableAllocator.
Commit3 removes PlanVariableAllocator
                       

```
== RELEASE NOTES ==

SPI Changes
* Added various methods to create variables in the VariableAllocator. VariableAllocator is also changed from interface to a class.
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVariableAllocator.java [hunk 1]
```diff
@@ -163,33 +80,11 @@ public TypeProvider getTypes()
...
     }
 
-    private int nextId()
-    {
-        return nextId++;
-    }
-
     public VariableReferenceExpression toVariableReference(Expression expression)
     {
-        checkArgument(expression instanceof SymbolReference, "Unexpected expression: %s", expression);
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVariableAllocator.java:173]: why did you change the %s?

Comment (jainxrohit) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVariableAllocator.java:173]: Not using guava methods anymore, I created my own utility

Comment (jainxrohit) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVariableAllocator.java:173]: Did not want to add guava dependency on spi

**Review by rschlussel — APPROVED** [review]
